broker/core/sql/src/database_config.cc

```diff
@@ -93,97 +101,144 @@ database_config::database_config(const std::string& type,
+      _config_logger->debug(
+          "No vault configuration file provided in Broker configuration.");
+    }
+  }
+  bool verify_peer = true;
+  {
+    auto found = global_params.find("verify_vault_peer");
+    if (found != global_params.end()) {
+      if (absl::SimpleAtob(found->second, &verify_peer)) {
+        _config_logger->debug("Verify Vault peer {}.",
+                              verify_peer ? "enabled" : "disabled");
+      } else {
+        _config_logger->debug("Verification of Vault peer enabled by default.");
+        verify_peer = true;
+      }
+    } else {
+      _config_logger->debug("Verification of Vault peer enabled by default.");
+    }
+  }
 
   // db_type
-  it = cfg.params.find("db_type");
-  if (it != end)
-    _type = it->second;
+  auto found = cfg.params.find("db_type");
+  if (found != cfg.params.end())
+    _type = found->second;
   else
     throw exceptions::config("no 'db_type' defined for endpoint '{}'",
                              cfg.name);
 
   // db_host
-  it = cfg.params.find("db_host");
-  if (it != end)
-    _host = it->second;
+  found = cfg.params.find("db_host");
+  if (found != cfg.params.end())
+    _host = found->second;
   else
     _host = "localhost";
 
   // db_socket
   if (_host == "localhost") {
-    it = cfg.params.find("db_socket");
-    if (it != end)
-      _socket = it->second;
+    found = cfg.params.find("db_socket");
+    if (found != cfg.params.end())
+      _socket = found->second;
     else
       _socket = MYSQL_SOCKET;
   } else
     _socket = "";
 
   // db_port
-  it = cfg.params.find("db_port");
-  auto logger_config = log_v2::instance().get(log_v2::CONFIG);
-  if (it != end) {
+  found = cfg.params.find("db_port");
+  if (found != cfg.params.end()) {
     uint32_t port;
-    if (!absl::SimpleAtoi(it->second, &port)) {
-      logger_config->error(
+    if (!absl::SimpleAtoi(found->second, &port)) {
+      _config_logger->error(
           "In the database configuration, 'db_port' should be a number, "
-          "and "
-          "not '{}'",
-          it->second);
+          "and not '{}'",
+          found->second);
       _port = 0;
     } else
       _port = port;
   } else
     _port = 0;
 
   // db_user
-  it = cfg.params.find("db_user");
-  if (it != end)
-    _user = it->second;
+  found = cfg.params.find("db_user");
+  if (found != cfg.params.end())
+    _user = found->second;
 
   // db_password
-  it = cfg.params.find("db_password");
-  if (it != end)
-    _password = it->second;
+  found = cfg.params.find("db_password");
+  if (found != cfg.params.end())
+    _password = found->second;
+
+  try {
+    common::vault::vault_access vault(env_file, vault_file, verify_peer,
+                                      _config_logger);
+    _password = vault.decrypt(_password);
+    _config_logger->info("Database password get from Vault configuration");
+  } catch (const std::exception& e) {
+    _config_logger->info("No usable Vault configuration: {}", e.what());
```

Comment:
Can we avoid to display these info in case of no vault is configured (most cases)

Reply:
I've added a filter with the password prefix.